splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/ModifyColumnConstantOperation.java

```diff
@@ -889,6 +889,25 @@ private void dropColumnFromTable(Activation activation,
         // list in case we were called recursively in order to cascade-drop a
         // dependent generated column.
         tab_cdl.remove( tableDescriptor.getColumnDescriptor( columnName ) );
+        ConglomerateDescriptor[] cds = tableDescriptor.getConglomerateDescriptors();
+        for (ConglomerateDescriptor cd : cds) {
+            boolean modified = false;
+            IndexDescriptor indexDescriptor =cd.getIndexDescriptor().getIndexDescriptor();
+            if (indexDescriptor != null) {
+                int[] positions = indexDescriptor.baseColumnLogicalPositions();
+                for (int i = 0; i < positions.length; ++i) {
+                    if (positions[i] > pos) {
+                        positions[i] = positions[i] - 1;
+                        modified = true;
+                    }
+                }
+                if (modified) {
+                    dd.dropConglomerateDescriptor(cd, tc);
+                    dd.addDescriptor(cd, sd,
+                            DataDictionary.SYSCONGLOMERATES_CATALOG_NUM, false, tc, false);
```

Comment:
It's a bit confusing here. Since we have modified the logical column positions in this `cd`, why would we drop it and add it back again? I can only image we do this to refresh data dictionary cache. But then it's kind of using its side effect rather than the actual effect.

If the purpose is to refresh cache only, can we achieve it without dropping and then adding it back? Or there is some other reason here?

Reply:
To update a conglomerate descriptor, we need to drop the old one from sysconglomerates first. Though this cd is new, only its schemaid and conglomerte name will be used to drop old conglomerates. The add this new conglomerate descriptor to dictionary